slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -319,28 +329,47 @@ bool SLAMServiceImpl::ExtractPointCloudToBuffer(std::stringbuf &buffer) {
 void SLAMServiceImpl::ProcessData() {
     // Set up and build the MapBuilder
     SetUpMapBuilder();
+    DetermineActionMode();
+
+    if (action_mode == SLAMServiceActionMode::UPDATING ||
+        action_mode == SLAMServiceActionMode::LOCALIZING) {
+        // Check if there is an apriori map in the path_to_map directory
+        std::vector<std::string> map_filenames =
+            viam::io::ListSortedFilesInDirectory(path_to_map);
+
+        bool found_map = false;
+        std::string latest_map_filename;
+        for (size_t i = map_filenames.size() - 1; i == 0; i--) {
+            if (map_filenames.at(i).find(".pbstream") != std::string::npos) {
```

Comment:
This ensures that if there's e.g. a DS_Store file present, that we're not counting it or trying to read it in.